### PR TITLE
Create menu dialog

### PR DESCRIPTION
**Pull request by proevan**
Create a class `MenuDialog` and move the code that relate to menu into this class

This PR is based on the PR #3652
please review starting from the commit: `Created MenuDialog that set up menu bottom bar and display menu items`

### CODE REVIEW

File: app/src/main/java/org/mozilla/rocket/extension/Extension.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package org.mozilla.rocket.extension
```
Comment (pharspher) [inline app/src/main/java/org/mozilla/rocket/extension/Extension.kt]: Would it be better to rename to something like ContextExtenstion or ContextExt, etc? Or this file is planned to contain extensions of all kinds of classes?

File: app/src/main/java/org/mozilla/rocket/extension/Extension.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+import android.app.Activity
+import android.content.Context
+import android.content.ContextWrapper
+import android.support.v4.app.FragmentActivity
+
+
+fun Context.toActivity(): Activity = when {
+    this is Activity -> this
+    this is ContextWrapper -> this.baseContext.toActivity()
```
Comment (pharspher) [inline app/src/main/java/org/mozilla/rocket/extension/Extension.kt]: what about nested context wrapper whose ancestor is an Activity?

File: app/src/main/java/org/mozilla/rocket/extension/Extension.kt (hunk 1 above)
Comment (proevan) [reply app/src/main/java/org/mozilla/rocket/extension/Extension.kt]: updated to ContextExtension

**Comment by proevan** [comment]
merge into PR #3682

**Pull request closed without merging**